chain/blockValidator.go

```diff
@@ -130,7 +130,7 @@ func TransactionCheck(ctx context.Context, block *block.Block) error {
 			return fmt.Errorf("transaction block check failed: %v", err)
 		}
 		bvs.Commit()
-		if err := VerifyTransactionWithLedger(txn); err != nil {
+		if err := VerifyTransactionWithLedger(txn, block.Header.UnsignedHeader.Height); err != nil {
```

Comment:
We don't need to pass block height to `VerifyTransactionWithLedger` (if we need, we should pass ledger height, not block.height). We can get ledger height inside the function. Search for `DefaultLedger.Store.GetHeight()` in `VerifyTransactionWithLedger` and you will find it